server/asset/eth/coiner.go

```diff
@@ -0,0 +1,284 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+//go:build lgpl
+// +build lgpl
+
+package eth
+
+import (
+	"context"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"math/big"
+
+	dexeth "decred.org/dcrdex/dex/networks/eth"
+	"decred.org/dcrdex/server/asset"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+type swapCoinType uint8
+
+const (
+	sctInit swapCoinType = iota
+	sctRedeem
+)
+
+var _ asset.Coin = (*swapCoin)(nil)
+
+type swapCoin struct {
+	backend                    *Backend
+	contractAddr, counterParty common.Address
+	secret, secretHash         [32]byte
+	value                      uint64
+	gasPrice                   uint64
+	txid                       string
+	locktime                   int64
+	sct                        swapCoinType
+}
+
+// newSwapCoin creats a new swapCoin that stores and retrieves info about a
+// swap. It requires a coinID that is a txid type of the initial transaction
+// initializing or redeeming the swap. A txid type and not a swap type is
+// required because the contract will give us no useful information about the
+// swap before it is mined. Having the initial transaction allows us to track
+// it in the mempool. It also tells us all the data we need to confirm a tx
+// will do what we expect if mined and satisfies contract contraints. These
+// fields are verified the first time the Confirmations method is called, and
+// an error is returned then if something is different than expected. As such,
+// the swapCoin expects Confirmations to be called with confirmations
+// available at least once before the swap be trusted for swap initializations.
+func newSwapCoin(backend *Backend, coinID []byte, sct swapCoinType) (*swapCoin, error) {
```

Comment:
Seems clear and reasonable.  Just a couple docs typos: "creats" and "contraints"